Sources/Shared/Request+Album.swift

```diff
@@ -29,13 +29,71 @@ public typealias AlbumRequest = Request<Album>
 /// `Request` that returns an array of `Album` objects.
 public typealias AlbumListRequest = Request<[Album]>
 
+fileprivate enum Key {
+    static let name = "name"
+    static let description = "description"
+    static let password = "password"
+    static let privacy = "privacy"
+}
+
 public extension Request {
-    
     /// Returns a new request to fetch a specific album.
     ///
     /// - Parameter uri: The album's URI.
     /// - Returns: Returns a new `Request` for an individual album.
     static func albumRequest(for uri: String) -> Request {
         return Request(path: uri)
     }
+
+    /// Returns a new request for creating an album.
+    /// - Parameter userURI: The URI for the current user, and owner of the album being created.
+    /// - Parameter name: The name of the album being created.
+    /// - Parameter description: An optional description for the album.
+    /// - Parameter privacy: The privacy parameter as a String. If none is specified it defaults to "anyone".
+    /// - Parameter password: An optional password parameter, only required when privacy is set to "password".
+    static func createAlbumRequest(
+        userURI: String,
+        name: String,
+        description: String? = nil,
+        privacy: String? = VIMPrivacy_Public,
+        password: String? = nil
+    ) -> Request {
+        var parameters = [String: String]()
+        parameters[Key.name] = name
+        parameters[Key.description] = description
+
+        privacy.map { parameters[Key.privacy] = $0 }
```

Comment:
why the mapping here? Can't you just use the subscript directly how you did with `description`?

Reply:
It's probably not strictly necessary to make the privacy option optional, but you _can_ actually ask the server to create an album without setting its privacy in the request, and the server default will also be to set the album to "anybody". 

So here you can omit the parameter and we'll set the privacy to "anybody". Or you can set it to `nil` and the server will set it to "anybody." Both are valid.

In the end I'm mapping here because sending the privacy parameter isn't a requirement.

Reply:
So are you saying the parameter gets sent with a blank value if we set this to `nil`? I thought it would simply remove the key from the dictionary no?

Reply:
You're correct, the key is removed from the dictionary entirely and no value is sent. The server accepts this and applies the default "anyone" privacy.

Reply:
Rog makes a good point! I think I suggested the `map` since the original logic was `if let privacy = privacy { parameters[Key.privacy] = privacy }`, but I didn't think about the fact that this logic wasn't necessary at all...

Reply:
Should I just make the parameter non-optional and apply our default here, instead of relying on the server to apply the default? That would probably make things simpler here.

Reply:
I think I'd agree with that.

Reply:
Okay cool, that sounds good to me.